src/aselo-webchat.ts

```diff
@@ -0,0 +1,207 @@
+    window.alert(translationErrorMsg);
+    console.error(translationErrorMsg, err);
+    getChangeLanguageWebChat(manager)(defaultLanguage);
+  }
+}
+
+type ChannelAndManagerFn = (channel: Channel, manager: FlexWebChat.Manager) => void;
+
+const doWithChannel = (callback: ChannelAndManagerFn) => (manager: FlexWebChat.Manager) => {
+  const { channelSid } = manager.store.getState().flex.session;
+    manager
+      .chatClient.getChannelBySid(channelSid)
+      .then(channel => {
+        callback(channel, manager);
+      });
+}
+
+const unlockInput = (manager: FlexWebChat.Manager) => {
+  const { user } = manager.chatClient;
+  const { lockInput, ...attributes } = user.attributes as any;
+  user.updateAttributes(attributes);
+}
+
+// TODO: we shouldn't need to cast a Channel to EventEmitter
+type EventEmitter = {
+  once: (eventName: string, listener: Function) => void;
+};
+
+const setListenerToUnlockInput = async (channel: Channel, manager: FlexWebChat.Manager) => {
+  if (!channel) return;
+
+  const cb = () => {
+    // Re-enable input
+    unlockInput(manager)
+  }
+
+  // User is not alone in the channel (possible cause to enter this case is page reload)
+  const membersCount = await channel.getMembersCount();
+  if (membersCount > 1) {
+    cb();
+    return;
+  }
+  
+  // Adds an event listener that will run only once
+  (channel as Channel & EventEmitter).once("memberJoined", () => {
+    cb();
+  });
+}
+
+const setChannelOnCreateWebChat = doWithChannel(setListenerToUnlockInput);
+
+const setChannelAfterStartEngagement = doWithChannel((channel: Channel, manager: FlexWebChat.Manager) => {
+  setListenerToUnlockInput(channel, manager);
+
+  // Generate task by sending empty message (hidden from the UI above)
+  channel.sendMessage(translations[initialLanguage].AutoFirstMessage);
+})
+
+export const initWebchat = () => FlexWebChat.createWebChat(appConfig).then(webchat => {
+  const { manager } = webchat;
+  if (manager.chatClient) console.log('>>>> User Attributes: ', { attributes : manager.chatClient.user.attributes });
+  const changeLanguageWebChat = getChangeLanguageWebChat(manager);
+
+  changeLanguageWebChat(initialLanguage);
+
+  // If caller is waiting for a counselor to connect, disable input (default language)
+  if (manager.chatClient) setChannelOnCreateWebChat(manager);
+
+  // Disable greeting message as chatbot already includes one
+  FlexWebChat.MessagingCanvas.defaultProps.predefinedMessage = undefined;
+
+  // Set caller name to be 'You'  
+  FlexWebChat.MessagingCanvas.defaultProps.memberDisplayOptions = {
+    yourDefaultName: 'You',
+    yourFriendlyNameOverride: false,
+    theirFriendlyNameOverride: true,
+  };
+
+  // Hide message input and send button if disabledReason is not undefined
+  FlexWebChat.MessageInput.Content.remove('textarea', {
+    // if: props => manager.chatClient.user.attributes.lockInput,
+    if: props => typeof FlexWebChat.MessagingCanvas.defaultProps.inputDisabledReason !== 'undefined',
```

Comment:
This line should actually be
```
if: props => (manager.chatClient.user.attributes as any).lockInput,
```
in order for this to work as expected (input in not being hidden right now). Sadly we need that ugly type casting because of the User.attributes type
`(property) User.attributes: Object`
results in
`Property 'lockInput' does not exist on type 'Object'.ts(2339)`

Reply:
Fixing it!